### ISSUE TITLE
Installing nx cli globally problem

### ISSUE THREAD

**Issue by memee** [nrwl/nx#4311]
## Current Behavior

1. `npm install -g nx`
2.  
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @nrwl/tao@11.0.5.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

## Expected Behavior

1. I can install `nx` cli globally

This is a regression

## Steps to Reproduce

`npm install nx -g`

### Failure Logs

```
0 verbose cli [
0 verbose cli   '/usr/local/Cellar/node/15.3.0/bin/node',
0 verbose cli   '/usr/local/bin/npm',
0 verbose cli   'install',
0 verbose cli   '-g',
0 verbose cli   'nx'
0 verbose cli ]
1 info using npm@7.0.14
2 info using node@v15.3.0
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 1ms
5 timing config:load:builtin Completed in 1ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/Users/<project-dir>/.npmrc Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/Users/macio/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 5ms
19 verbose npm-session 4e5fb756e96725c7
20 timing npm:load Completed in 11ms
21 timing arborist:ctor Completed in 1ms
22 timing idealTree:init Completed in 5ms
23 timing idealTree:userRequests Completed in 1ms
24 silly idealTree buildDeps
25 silly fetch manifest nx@*
26 http fetch GET 304 https://registry.npmjs.org/nx 189ms (from cache)
27 silly placeDep ROOT nx@11.0.4 OK for:  want: *
28 silly fetch manifest @nrwl/cli@*
29 http fetch GET 304 https://registry.npmjs.org/@nrwl%2fcli 78ms (from cache)
30 timing idealTree:#root Completed in 277ms
31 silly placeDep node_modules/nx @nrwl/cli@11.0.5 OK for: nx@11.0.4 want: *
32 silly fetch manifest tmp@0.0.33
33 silly fetch manifest yargs@15.4.1
34 silly fetch manifest yargs-parser@20.0.0
35 silly fetch manifest @nrwl/tao@11.0.5
36 silly fetch manifest chalk@4.1.0
37 http fetch GET 304 https://registry.npmjs.org/tmp 65ms (from cache)
38 http fetch GET 304 https://registry.npmjs.org/chalk 188ms (from cache)
39 http fetch GET 304 https://registry.npmjs.org/yargs-parser 190ms (from cache)
40 http fetch GET 304 https://registry.npmjs.org/yargs 194ms (from cache)
41 http fetch GET 304 https://registry.npmjs.org/@nrwl%2ftao 261ms (from cache)
42 timing idealTree:node_modules/nx Completed in 268ms
43 silly placeDep node_modules/nx @nrwl/tao@ OK for: @nrwl/cli@11.0.5 want: 11.0.5
44 silly placeDep node_modules/nx chalk@4.1.0 OK for: @nrwl/cli@11.0.5 want: 4.1.0
45 silly placeDep node_modules/nx tmp@0.0.33 OK for: @nrwl/cli@11.0.5 want: 0.0.33
46 silly placeDep node_modules/nx yargs@15.4.1 OK for: @nrwl/cli@11.0.5 want: 15.4.1
47 silly placeDep node_modules/nx yargs-parser@20.0.0 OK for: @nrwl/cli@11.0.5 want: 20.0.0
48 silly fetch manifest ansi-styles@^4.1.0
49 silly fetch manifest supports-color@^7.1.0
50 silly fetch manifest os-tmpdir@~1.0.2
51 silly fetch manifest cliui@^6.0.0
52 silly fetch manifest decamelize@^1.2.0
53 silly fetch manifest find-up@^4.1.0
54 silly fetch manifest get-caller-file@^2.0.1
55 silly fetch manifest require-directory@^2.1.1
56 silly fetch manifest require-main-filename@^2.0.0
57 silly fetch manifest set-blocking@^2.0.0
58 silly fetch manifest string-width@^4.2.0
59 silly fetch manifest which-module@^2.0.0
60 silly fetch manifest y18n@^4.0.0
61 silly fetch manifest yargs-parser@^18.1.2
62 http fetch GET 304 https://registry.npmjs.org/os-tmpdir 168ms (from cache)
63 http fetch GET 304 https://registry.npmjs.org/get-caller-file 166ms (from cache)
64 http fetch GET 304 https://registry.npmjs.org/require-directory 165ms (from cache)
65 http fetch GET 304 https://registry.npmjs.org/decamelize 169ms (from cache)
66 http fetch GET 304 https://registry.npmjs.org/cliui 171ms (from cache)
67 http fetch GET 304 https://registry.npmjs.org/ansi-styles 179ms (from cache)
68 http fetch GET 304 https://registry.npmjs.org/supports-color 194ms (from cache)
69 http fetch GET 304 https://registry.npmjs.org/y18n 189ms (from cache)
70 http fetch GET 304 https://registry.npmjs.org/find-up 197ms (from cache)
71 http fetch GET 304 https://registry.npmjs.org/require-main-filename 197ms (from cache)
72 http fetch GET 304 https://registry.npmjs.org/set-blocking 197ms (from cache)
73 http fetch GET 304 https://registry.npmjs.org/which-module 196ms (from cache)
74 http fetch GET 304 https://registry.npmjs.org/yargs-parser 214ms (from cache)
75 http fetch GET 304 https://registry.npmjs.org/string-width 220ms (from cache)
76 timing idealTree:node_modules/nx/node_modules/@nrwl/cli Completed in 236ms
77 timing idealTree:node_modules/nx/node_modules/@nrwl/tao Completed in 0ms
78 silly placeDep node_modules/nx ansi-styles@4.3.0 OK for: chalk@4.1.0 want: ^4.1.0
79 silly placeDep node_modules/nx supports-color@7.2.0 OK for: chalk@4.1.0 want: ^7.1.0
80 silly fetch manifest color-convert@^2.0.1
81 silly fetch manifest has-flag@^4.0.0
82 http fetch GET 304 https://registry.npmjs.org/has-flag 63ms (from cache)
83 http fetch GET 304 https://registry.npmjs.org/color-convert 92ms (from cache)
84 timing idealTree:node_modules/nx/node_modules/chalk Completed in 95ms
85 silly placeDep node_modules/nx color-convert@2.0.1 OK for: ansi-styles@4.3.0 want: ^2.0.1
86 silly fetch manifest color-name@~1.1.4
87 http fetch GET 304 https://registry.npmjs.org/color-name 60ms (from cache)
88 timing idealTree:node_modules/nx/node_modules/ansi-styles Completed in 61ms
89 silly placeDep node_modules/nx color-name@1.1.4 OK for: color-convert@2.0.1 want: ~1.1.4
90 timing idealTree:node_modules/nx/node_modules/color-convert Completed in 0ms
91 timing idealTree:node_modules/nx/node_modules/color-name Completed in 0ms
92 silly placeDep node_modules/nx has-flag@4.0.0 OK for: supports-color@7.2.0 want: ^4.0.0
93 timing idealTree:node_modules/nx/node_modules/supports-color Completed in 1ms
94 timing idealTree:node_modules/nx/node_modules/has-flag Completed in 0ms
95 silly placeDep node_modules/nx os-tmpdir@1.0.2 OK for: tmp@0.0.33 want: ~1.0.2
96 timing idealTree:node_modules/nx/node_modules/tmp Completed in 1ms
97 timing idealTree:node_modules/nx/node_modules/os-tmpdir Completed in 0ms
98 silly placeDep node_modules/nx cliui@6.0.0 OK for: yargs@15.4.1 want: ^6.0.0
99 silly placeDep node_modules/nx decamelize@1.2.0 OK for: yargs@15.4.1 want: ^1.2.0
100 silly placeDep node_modules/nx find-up@4.1.0 OK for: yargs@15.4.1 want: ^4.1.0
101 silly placeDep node_modules/nx get-caller-file@2.0.5 OK for: yargs@15.4.1 want: ^2.0.1
102 silly placeDep node_modules/nx require-directory@2.1.1 OK for: yargs@15.4.1 want: ^2.1.1
103 silly placeDep node_modules/nx require-main-filename@2.0.0 OK for: yargs@15.4.1 want: ^2.0.0
104 silly placeDep node_modules/nx set-blocking@2.0.0 OK for: yargs@15.4.1 want: ^2.0.0
105 silly placeDep node_modules/nx string-width@4.2.0 OK for: yargs@15.4.1 want: ^4.2.0
106 silly placeDep node_modules/nx which-module@2.0.0 OK for: yargs@15.4.1 want: ^2.0.0
107 silly placeDep node_modules/nx y18n@4.0.1 OK for: yargs@15.4.1 want: ^4.0.0
108 silly placeDep node_modules/nx/node_modules/yargs yargs-parser@18.1.3 OK for: yargs@15.4.1 want: ^18.1.2
109 silly fetch manifest strip-ansi@^6.0.0
110 silly fetch manifest wrap-ansi@^6.2.0
111 silly fetch manifest locate-path@^5.0.0
112 silly fetch manifest path-exists@^4.0.0
113 silly fetch manifest emoji-regex@^8.0.0
114 silly fetch manifest is-fullwidth-code-point@^3.0.0
115 silly fetch manifest camelcase@^5.0.0
116 http fetch GET 304 https://registry.npmjs.org/path-exists 90ms (from cache)
117 http fetch GET 304 https://registry.npmjs.org/locate-path 92ms (from cache)
118 http fetch GET 304 https://registry.npmjs.org/emoji-regex 90ms (from cache)
119 http fetch GET 304 https://registry.npmjs.org/strip-ansi 97ms (from cache)
120 http fetch GET 304 https://registry.npmjs.org/wrap-ansi 97ms (from cache)
121 http fetch GET 304 https://registry.npmjs.org/is-fullwidth-code-point 93ms (from cache)
122 http fetch GET 304 https://registry.npmjs.org/camelcase 107ms (from cache)
123 timing idealTree:node_modules/nx/node_modules/yargs Completed in 121ms
124 silly placeDep node_modules/nx strip-ansi@6.0.0 OK for: cliui@6.0.0 want: ^6.0.0
125 silly placeDep node_modules/nx wrap-ansi@6.2.0 OK for: cliui@6.0.0 want: ^6.2.0
126 silly fetch manifest ansi-regex@^5.0.0
127 http fetch GET 304 https://registry.npmjs.org/ansi-regex 56ms (from cache)
128 timing idealTree:node_modules/nx/node_modules/cliui Completed in 59ms
129 timing idealTree:node_modules/nx/node_modules/decamelize Completed in 0ms
130 silly placeDep node_modules/nx locate-path@5.0.0 OK for: find-up@4.1.0 want: ^5.0.0
131 silly placeDep node_modules/nx path-exists@4.0.0 OK for: find-up@4.1.0 want: ^4.0.0
132 silly fetch manifest p-locate@^4.1.0
133 http fetch GET 304 https://registry.npmjs.org/p-locate 44ms (from cache)
134 timing idealTree:node_modules/nx/node_modules/find-up Completed in 46ms
135 timing idealTree:node_modules/nx/node_modules/get-caller-file Completed in 0ms
136 silly placeDep node_modules/nx p-locate@4.1.0 OK for: locate-path@5.0.0 want: ^4.1.0
137 silly fetch manifest p-limit@^2.2.0
138 http fetch GET 304 https://registry.npmjs.org/p-limit 42ms (from cache)
139 timing idealTree:node_modules/nx/node_modules/locate-path Completed in 44ms
140 silly placeDep node_modules/nx p-limit@2.3.0 OK for: p-locate@4.1.0 want: ^2.2.0
141 silly fetch manifest p-try@^2.0.0
142 http fetch GET 304 https://registry.npmjs.org/p-try 37ms (from cache)
143 timing idealTree:node_modules/nx/node_modules/p-locate Completed in 40ms
144 silly placeDep node_modules/nx p-try@2.2.0 OK for: p-limit@2.3.0 want: ^2.0.0
145 timing idealTree:node_modules/nx/node_modules/p-limit Completed in 1ms
146 timing idealTree:node_modules/nx/node_modules/p-try Completed in 0ms
147 timing idealTree:node_modules/nx/node_modules/path-exists Completed in 0ms
148 timing idealTree:node_modules/nx/node_modules/require-directory Completed in 0ms
149 timing idealTree:node_modules/nx/node_modules/require-main-filename Completed in 0ms
150 timing idealTree:node_modules/nx/node_modules/set-blocking Completed in 0ms
151 silly placeDep node_modules/nx emoji-regex@8.0.0 OK for: string-width@4.2.0 want: ^8.0.0
152 silly placeDep node_modules/nx is-fullwidth-code-point@3.0.0 OK for: string-width@4.2.0 want: ^3.0.0
153 timing idealTree:node_modules/nx/node_modules/string-width Completed in 2ms
154 timing idealTree:node_modules/nx/node_modules/emoji-regex Completed in 0ms
155 timing idealTree:node_modules/nx/node_modules/is-fullwidth-code-point Completed in 0ms
156 silly placeDep node_modules/nx ansi-regex@5.0.0 OK for: strip-ansi@6.0.0 want: ^5.0.0
157 timing idealTree:node_modules/nx/node_modules/strip-ansi Completed in 1ms
158 timing idealTree:node_modules/nx/node_modules/ansi-regex Completed in 0ms
159 timing idealTree:node_modules/nx/node_modules/which-module Completed in 0ms
160 timing idealTree:node_modules/nx/node_modules/wrap-ansi Completed in 0ms
161 timing idealTree:node_modules/nx/node_modules/y18n Completed in 0ms
162 timing idealTree:node_modules/nx/node_modules/yargs-parser Completed in 0ms
163 silly placeDep node_modules/nx camelcase@5.3.1 OK for: yargs-parser@18.1.3 want: ^5.0.0
164 timing idealTree:node_modules/nx/node_modules/yargs/node_modules/yargs-parser Completed in 1ms
165 timing idealTree:node_modules/nx/node_modules/camelcase Completed in 0ms
166 timing idealTree:buildDeps Completed in 1256ms
167 timing idealTree:fixDepFlags Completed in 0ms
168 timing idealTree Completed in 1265ms
169 timing command:install Completed in 1267ms
170 verbose type version
171 verbose stack @nrwl/tao: No matching version found for @nrwl/tao@11.0.5.
171 verbose stack     at module.exports (/usr/local/lib/node_modules/npm/node_modules/npm-pick-manifest/index.js:207:23)
171 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/pacote/lib/registry.js:96:26
171 verbose stack     at async Promise.all (index 3)
171 verbose stack     at async Arborist.[buildDepStep] (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:811:5)
171 verbose stack     at async Arborist.buildIdealTree (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:209:7)
171 verbose stack     at async Arborist.reify (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:122:5)
171 verbose stack     at async install (/usr/local/lib/node_modules/npm/lib/install.js:39:3)
172 verbose cwd /Users/<project>
173 verbose Darwin 19.6.0
174 verbose argv "/usr/local/Cellar/node/15.3.0/bin/node" "/usr/local/bin/npm" "install" "-g" "nx"
175 verbose node v15.3.0
176 verbose npm  v7.0.14
177 error code ETARGET
178 error notarget No matching version found for @nrwl/tao@11.0.5.
179 error notarget In most cases you or one of your dependencies are requesting
179 error notarget a package version that doesn't exist.
180 verbose exit 1
```

### Environment

```
nx report

>  NX  Report complete - copy this into the issue template

  nx : Not Found
  @nrwl/angular : 10.3.3
  @nrwl/cli : 10.3.0
  @nrwl/cypress : 10.3.3
  @nrwl/eslint-plugin-nx : Not Found
  @nrwl/express : Not Found
  @nrwl/jest : 10.3.3
  @nrwl/linter : 10.3.3
  @nrwl/nest : 10.3.3
  @nrwl/next : Not Found
  @nrwl/node : 10.3.3
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.3.0
  @nrwl/web : Not Found
  @nrwl/workspace : 10.3.3
  typescript : 4.0.3
```

**Comment by vinityad** [comment]
Getting same issue with Azure DevOps pipeline and now in local machine too. Tried installing @nrwl/tao@11.0.4 (because there is no version 11.0.5) manually but no luck.

**Comment by Flintlozk** [comment]
Have the same problem on the latest version. So I prevent this by forces the version of @nrwl/cli (10.4.4).
```
npm i -g @nrwl/cli                                            
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @nrwl/tao@11.0.5.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of '@nrwl/cli'
npm ERR! notarget```

**Comment by vsavkin** [comment]
Really sorry about it, folks. Fixed in 11.0.8.

**Comment by arilaen** [comment]
Am seeing this again after the nx 12.0.2 release, guessing that CLI was updated as well since npm i -g nx for any version I've tested is resulting in the same error above.
